ethereum/api/src/test-support/java/org/hyperledger/besu/ethereum/api/tls/SelfSignedPfxStore.java

```diff
@@ -0,0 +1,241 @@
+    final X500Name subject = new X500Name(distinguishedName);
+    final BigInteger serialNumber = new BigInteger(String.valueOf(Instant.now().toEpochMilli()));
+    final X509v3CertificateBuilder v3CertificateBuilder =
+        new JcaX509v3CertificateBuilder(
+            issuer,
+            serialNumber,
+            Date.from(Instant.now()),
+            Date.from(Instant.now().plus(Period.ofDays(90))),
+            subject,
+            keyPair.getPublic());
+
+    // extensions
+    v3CertificateBuilder.addExtension(
+        Extension.basicConstraints, true, new BasicConstraints(IS_CA));
+    v3CertificateBuilder.addExtension(
+        Extension.subjectAlternativeName, false, getSubjectAlternativeNames());
+
+    final ContentSigner contentSigner =
+        new JcaContentSignerBuilder("SHA256WithRSAEncryption").build(keyPair.getPrivate());
+
+    this.certificate =
+        new JcaX509CertificateConverter()
+            .setProvider(BOUNCY_CASTLE_PROVIDER)
+            .getCertificate(v3CertificateBuilder.build(contentSigner));
+  }
+
+  private GeneralNames getSubjectAlternativeNames() {
+    final List<GeneralName> hostGeneralNames =
+        sanHostNames.stream()
+            .map(hostName -> new GeneralName(GeneralName.dNSName, hostName))
+            .collect(Collectors.toList());
+    final List<GeneralName> ipGeneralNames =
+        sanIpAddresses.stream()
+            .map(ipAddress -> new GeneralName(GeneralName.iPAddress, ipAddress))
+            .collect(Collectors.toList());
+    final GeneralName[] generalNames =
+        Stream.of(hostGeneralNames, ipGeneralNames)
+            .flatMap(Collection::stream)
+            .toArray(GeneralName[]::new);
+
+    return new GeneralNames(generalNames);
+  }
+
+  private void createKeyStore() throws IOException, GeneralSecurityException {
+    this.keyStore = convert(true);
+  }
+
+  private void createTrustStore() throws IOException, GeneralSecurityException {
+    this.trustStore = convert(false);
+  }
+
+  private Path convert(final boolean isKeyStore) throws IOException, GeneralSecurityException {
+    final KeyStore keyStore = KeyStore.getInstance("PKCS12");
+    keyStore.load(null);
+    if (isKeyStore) {
+      keyStore.setKeyEntry(alias, keyPair.getPrivate(), password, new Certificate[] {certificate});
+    } else {
+      keyStore.setCertificateEntry(alias, certificate);
+    }
+
+    return saveKeyStore(keyStore);
+  }
+
+  private Path saveKeyStore(final KeyStore keyStore)
+      throws IOException, KeyStoreException, NoSuchAlgorithmException, CertificateException {
+    final Path pfxPath = Files.createTempFile(parentPath, alias, ".pfx");
+    pfxPath.toFile().deleteOnExit();
+    try (final FileOutputStream outputStream = new FileOutputStream(pfxPath.toFile())) {
+      keyStore.store(outputStream, password);
+    }
+    return pfxPath;
+  }
+
+  private void createKnownClientsFile() throws IOException, CertificateEncodingException {
```

Comment:
maybe this doesn't have a known-clients file, but rather, allows you to get the fingerprint of the certificate, that way the user of this class can choose what to do with the fingerprint - i.e. compare it, write it to file etc.etc.

Reply:
Extracted out to a different utility class. It now determines common name and fingerprint dynamically from trust store.